### PR TITLE
debug_traceTransaction with JSON streaming

### PR DESCRIPTION
**Pull request by Sixtysixter**
JSON streaming for debug_traceTransaction API

### CODE REVIEW

File: silkrpc/json/stream.cpp [hunk 1]
```diff
@@ -16,12 +16,94 @@
...
+#include <string>
+#include <vector>
+
+#include <boost/asio/write.hpp>
+
 namespace json {
-boost::asio::awaitable<void> Stream::flush() {
-    co_return;
+
+static std::uint8_t OBJECT_OPEN = 1;
```
Comment (canepat) [inline silkrpc/json/stream.cpp:27]: Rename `OBJECT_OPEN` as `kObjectOpen` to follow naming conventions

File: silkrpc/json/stream_test.cpp [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+    //         "test": "test"
+    //     })"_json;
+
+    //     const auto content = json.dump(/*indent=*/-1, /*indent_char=*/' ', /*ensure_ascii=*/false, nlohmann::json::error_handler_t::replace);
+
+    //     stream.write_json(content);
+    //     stream.close();
+
+    //     CHECK(string_writer.get_content() == "{\"test\":\"test\"}");
+    // }
```
Comment (canepat) [inline silkrpc/json/stream_test.cpp:100]: Remove commented test

**Review by canepat — CHANGES_REQUESTED** [review]

File: silkrpc/commands/rpc_api_table.cpp [hunk 3]
```diff
@@ -84,10 +84,12 @@ void RpcApiTable::add_debug_handlers() {
     method_handlers_[http::method::k_debug_getModifiedAccountsByNumber] = &commands::RpcApi::handle_debug_get_modified_accounts_by_number;
     method_handlers_[http::method::k_debug_getModifiedAccountsByHash] = &commands::RpcApi::handle_debug_get_modified_accounts_by_hash;
     method_handlers_[http::method::k_debug_storageRangeAt] = &commands::RpcApi::handle_debug_storage_range_at;
-    method_handlers_[http::method::k_debug_traceTransaction] = &commands::RpcApi::handle_debug_trace_transaction;
+    // method_handlers_[http::method::k_debug_traceTransaction] = &commands::RpcApi::handle_debug_trace_transaction;
```
[the remark this answers is not in the document]
Comment (Sixtysixter) [reply silkrpc/commands/rpc_api_table.cpp:87]: Done

File: silkrpc/http/request_handler.cpp [hunk 4]
```diff
@@ -162,13 +159,16 @@ boost::asio::awaitable<void> RequestHandler::handle_request(silkrpc::commands::R
...
-    auto request_id = request_json["id"].get<uint32_t>();
+boost::asio::awaitable<void> RequestHandler::handle_request(silkrpc::commands::RpcApiTable::HandleStream handler, const nlohmann::json& request_json) {
     try {
-        json::Stream stream(socket_);
+        SocketWriter socket_writer(socket_);
+        ChunksWriter chunks_writer(socket_writer);
+        json::Stream stream(chunks_writer);
 
-        co_await write_headers();
+        co_await start_streaming();
```
[the remark this answers is not in the document]
Comment (Sixtysixter) [reply silkrpc/http/request_handler.cpp:168]: Done

File: silkrpc/json/stream.cpp (hunk 1 above)
Comment (Sixtysixter) [reply silkrpc/json/stream.cpp:27]: OK, done

File: silkrpc/json/stream_test.cpp (hunk 2 above)
Comment (Sixtysixter) [reply silkrpc/json/stream_test.cpp:100]: Done

**Review by canepat — APPROVED** [review]

**Pull request merged**